File: javascript/Day_4/Day_4.js
/**
 * Problème
 * --------
 * Concevoir un programme qui demande l'âge d'une personne.
 * Ensuite, l'informe de sa catégorie :
 * "Enfance : de 0 à 12 ans",
 * "Adolescence : de 12 à 18 ans",
 * "Âge Adulte : de 18 à 65 ans",
 * "Séniors ou Personnes âgées : À partir de 65 ans"
 *
 * Catégorie
 * ---------
 * Numbers
 *
 * Niveau
 * ------
 * Débutant
 *
 * @author 10Glo
 * @version 1.0
 * @since 16/09/2025
 */

// --- Dépendance pour la saisie ---
const prompt = require("prompt-sync")({ sigint: true });

// --- Déclaration des Variables ---
let age = 0;

// --- Titre du projet ---
console.log("==============================================");
console.log(":::::::::      :::   :::   :::         :::     ");
console.log(":+:    :+:   :+: :+: :+:   :+:        :+:      ");
console.log("+:+    +:+  +:+   +:+ +:+ +:+        +:+ +:+   ");
console.log("+#+    +:+ +#++:++#++: +#++:        +#+  +:+   ");
console.log("+#+    +#+ +#+     +#+  +#+        +#+#+#+#+#+ ");
console.log("#+#    #+# #+#     #+#  #+#              #+#   ");
console.log("#########  ###     ###  ###              ###   ");
console.log("==============================================\n");

// --- Saisie de l'âge ---
try {
    let input = prompt("➡️  Entrez votre âge (Entrez juste le nombre) : ");
    age = parseInt(input);

    if (isNaN(age)) {
        console.log("\n⚠️ Erreur : Vous devez entrer un nombre valide !");
        console.log("\nFin du programme ✅");
        process.exit(0);
    }
} catch (error) {
    console.log("\n⚠️ Erreur : Impossible de lire la saisie !");
    console.log("\nFin du programme ✅");
    process.exit(0);
}

// --- Résumé des informations ---
console.log("\n-------------------------------------");
console.log(" 📋 Résumé des informations saisies");
console.log("-------------------------------------");
console.log(" 🔢 L'âge que vous avez entré : " + age + " ans\n");

// --- Résultats ---
console.log("--------------");
console.log(" 📋 Résultats");
console.log("--------------");

// --- Règles ---
if (age < 0) {
    console.log("Votre âge ne peut pas être négatif.\n");
} else if (age <= 12) {
    console.log("➡️  Votre catégorie est 'ENFANT'\n");
} else if (age <= 18) {
    console.log("➡️  Votre catégorie est 'ADOLESCENCE'\n");
} else if (age <= 65) {
    console.log("➡️  Votre catégorie est 'ADULTE'\n");
} else {
    console.log("➡️  Votre catégorie est 'SÉNIORS'\n");
}

// --- Message de fin ---
console.log("Fin du programme ✅");
console.log("-------------------------------------------");

File: java/Day_4/src/Day_4.java
import java.util.Scanner;

/**
 * Problème
 * --------
 * Concevoir un programme qui demande l'age d'une personne.
 * Ensuite, l'informe de sa catégorie ("Enfance : de 0 à 12ans", "Adolescence : De 12 à 18 ans,"
 * "Âge Adulte : De 18 à 65 ans" et "Séniors ou Personnes âgées : À partir de 65 ans")
 *
 * Catégorie
 * ---------
 * Numbers
 *
 * Niveau
 * ------
 * Débutant
 *
 * @author 10Glo (https://github.com/10Glo)
 * @version 1.0
 * @since 16/09/2025
 */

public class Day_4 {

    public static void main(String[] args) {

        // --- Déclaration des Variables ---
        Scanner scanner = new Scanner(System.in);
        int age = 0;

        // --- Titre du projet ---
        System.out.println("==============================================");
        System.out.println(":::::::::      :::   :::   :::         :::     \n" +
                ":+:    :+:   :+: :+: :+:   :+:        :+:      \n" +
                "+:+    +:+  +:+   +:+ +:+ +:+        +:+ +:+   \n" +
                "+#+    +:+ +#++:++#++: +#++:        +#+  +:+   \n" +
                "+#+    +#+ +#+     +#+  +#+        +#+#+#+#+#+ \n" +
                "#+#    #+# #+#     #+#  #+#              #+#   \n" +
                "#########  ###     ###  ###              ###   ");
        System.out.println("==============================================\n");

        // --- Saisie de l'âge ---
        try {
            System.out.print("➡️  Entrez votre âge (Entrez juste le nombre) : ");
            age = scanner.nextInt();
        } catch (Exception exception) {
            System.out.println("\n⚠️ Erreur : Vous devez entrer un nombre valide !");

            scanner.close();
            System.out.println("\nFin du programme ✅");
            System.exit(0);
        }

        // --- Résumé des informations ---
        System.out.println("\n-------------------------------------");
        System.out.println(" 📋 Résumé des informations saisies");
        System.out.println("-------------------------------------");
        System.out.println(" \uD83D\uDD22 L'âge que vous avez entré : " + age + " ans\n");

        // --- Résutats ---
        System.out.println("--------------");
        System.out.println(" 📋 Résutats");
        System.out.println("--------------");

        // --- Régles ---
        if (age < 0)
            System.out.println("Votre âge ne peux être négatif'\n");
        else if (age <= 12)
            System.out.println("➡️  Votre catégorie est 'ENFANT'\n");
        else if (age <= 18)
            System.out.println("➡️  Votre catégorie est 'ADOLESCENCE'\n");
        else if (age <= 65)
            System.out.println("➡️  Votre catégorie est 'ADULTE'\n");
        else
            System.out.println("➡️  Votre catégorie est 'SÉNIORS'\n");

        // --- Message de fin ---
        scanner.close();
        System.out.println("Fin du programme ✅");
        System.out.println("-------------------------------------------");


    }
}
